integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/outbox/CustomRoutingBridge.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.mapper.orm.outbox;
+
+import org.hibernate.search.mapper.pojo.bridge.RoutingBridge;
+import org.hibernate.search.mapper.pojo.bridge.binding.RoutingBindingContext;
+import org.hibernate.search.mapper.pojo.bridge.mapping.programmatic.RoutingBinder;
+import org.hibernate.search.mapper.pojo.bridge.runtime.RoutingBridgeRouteContext;
+import org.hibernate.search.mapper.pojo.route.DocumentRoutes;
+
+public class CustomRoutingBridge implements RoutingBridge<RoutedIndexedEntity> {
+
+	public static final int NUMBER_OF_SHARD = 7;
+
+	@Override
+	public void route(DocumentRoutes routes, Object entityIdentifier, RoutedIndexedEntity indexedEntity,
+			RoutingBridgeRouteContext context) {
+		String text = indexedEntity.getText();
+		char selector;
+
+		if ( text == null || text.trim().isEmpty() ) {
+			selector = 'z';
+		}
+		else {
+			selector = text.charAt( 0 );
+		}
+
+		// this is not a really great way to define the route id,
+		// we're using here just for testing
+		int shardId = selector % NUMBER_OF_SHARD;
+		routes.addRoute().routingKey( shardId + "" );
```

Comment:
Sharding and the number of shards should not be a concern here. Those are configured on the backend side.

This would work just as well (especially for tests), and it's much simpler:

```suggestion
		routes.addRoute().routingKey( indexedEntity.getSomeEnumProperty().name() );
```